shoop/admin/modules/orders/json_order_creator.py

```diff
@@ -121,27 +136,93 @@ def _process_lines(self, source, state):
-        customer_data = state.pop("customer", None) or {}
-        shop_data = state.pop("shop", None) or {}
-        methods_data = state.pop("methods", None) or {}
-        customer = self.safe_get_first(Contact, pk=customer_data.get("id"))
-        shop = self.safe_get_first(Shop, pk=shop_data.pop("id", None))
+    def _create_contact_from_address(self, billing_address, is_company):
+        name = billing_address.get("name", None)
+        if not name:
+            self.add_error(
+                ValidationError(_("Name is required for customer"), code="no_name")
+            )
 
-        if not customer:
-            self.add_error(ValidationError(_("Please choose a valid customer."), code="no_customer"))
+        phone = billing_address.get("phone", "")
+        email = billing_address.get("email", "")
+        fields = {"name": name, "phone": phone, "email": email}
+        if is_company:
+            tax_number = billing_address.pop("tax_number", None)
+            if not tax_number:
+                self.add_error(
+                    ValidationError(_("Tax number is not set for company."), code="no_tax_number")
+                )
+                return
+
+            fields.update({"tax_number": tax_number})
+            customer = CompanyContact.objects.create(**fields)
+        else:
+            customer = PersonContact.objects.create(**fields)
+        return customer
 
+    def _create_address(self, address):
+        address_fields = [
+            "name", "tax_number", "phone", "email", "street", "street2", "postal_code", "city", "region", "country"
+        ]
+        fields = {}
+        for field in address_fields:
+            fields[field] = address.get(field, "")
+
+        address = Address.objects.filter(**fields).first()
+        if not address:
+            address = Address.objects.create(**fields)
+
+        return address
+
+    def _initialize_source_from_state(self, state, creator):
+        shop_data = state.pop("shop", None).get("selected", {})
+        shop = self.safe_get_first(Shop, pk=shop_data.pop("id", None))
         if not shop:
             self.add_error(ValidationError(_("Please choose a valid shop."), code="no_shop"))
             return None
 
         source = OrderSource(shop=shop)
+
+        customer_data = state.pop("customer", None)
+        billing_address = customer_data.pop("billingAddress", {})
+        shipping_address = customer_data.pop("shippingAddress", {})
+        ship_to_billing_address = customer_data.pop("shipToBillingAddress", False)
+        is_company = customer_data.pop("isCompany", False)
+        save_address = customer_data.pop("saveAddress", False)
+        customer = self.safe_get_first(Contact, pk=customer_data.get("id")) if customer_data else None
+        if not customer:
+            customer = self._create_contact_from_address(billing_address, is_company)
+            if not customer:
+                return
```

Comment:
Creating contact might be skipped if there is no tax_number for company. In that case this return ends order creation, which should be fine.